src/com/ms/silverking/cloud/dht/benchmark/ycsb/GroupingSilverkingDB.java

```diff
@@ -32,173 +31,175 @@
-            //}
-            if (System.currentTimeMillis() == 0) {
-                syncNSP.put(key, values);
-            }
-            /**/
-            {
-                AsyncPut    asyncPut;
-                
-                //System.out.println("Waiting for: "+ key);
-                asyncPut = asyncNSP.put(key, values);
-                try {
-                    asyncPut.waitForCompletion(30, TimeUnit.SECONDS);
-                    if (asyncPut.getState() == OperationState.INCOMPLETE) {
-                        System.out.println("Incomplete: "+ key);
-                        System.out.flush();
-                        System.exit(-1);
-                    }
-                } catch (OperationException e) {
-                    PutException    pe;
-                    
-                    pe = (PutException)e;
-                    if (pe.getFailureCause(key) == FailureCause.INVALID_VERSION) {
-                        System.out.println("Ignoring INVALID_VERSION");
-                        System.err.println("Ignoring INVALID_VERSION");
-                        return 1;
-                    } else {
-                        throw new RuntimeException(pe);
-                    }
-                }
-            }
-            /**/
-            //System.out.println("Done waiting for: "+ key);
-            return 0;
-        } catch (PutException pe) {
-            System.out.println("Key failed: "+ key);
+    //version = new AtomicInteger(((int)(System.currentTimeMillis() * 10000) >>> 2) & 0x3fffffff);
+    version = new AtomicLong(System.nanoTime());
+  }
+
+  public void init() throws DBException {
+    try {
+      if (session == null) {
+        throw new RuntimeException("null session");
+      }
+      //syncNSP = session.openSyncNamespacePerspective(SilverkingDBConstants.defaultNamespace, SilverkingDBConstants
+      // .nspOptions);
+      //asyncNSP = session.openAsyncNamespacePerspective(SilverkingDBConstants.defaultNamespace,
+      // SilverkingDBConstants.nspOptions);
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public int insert(String table, String key, HashMap<String, ByteIterator> values) {
+    try {
+      //for (Map.Entry<String, ByteIterator> entry : values.entrySet()) {
+      //    System.out.println(entry.getKey() +"\t"+ entry.getValue().toString());
+      //}
+      if (System.currentTimeMillis() == 0) {
+        syncNSP.put(key, values);
+      }
+      /**/
+      {
+        AsyncPut asyncPut;
+
+        //System.out.println("Waiting for: "+ key);
+        asyncPut = asyncNSP.put(key, values);
+        try {
+          asyncPut.waitForCompletion(30, TimeUnit.SECONDS);
+          if (asyncPut.getState() == OperationState.INCOMPLETE) {
+            System.out.println("Incomplete: " + key);
+            System.out.flush();
+            System.exit(-1);
+          }
+        } catch (OperationException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)